src/Docfx.Build.Engine/DocumentBuildContext.cs

```diff
@@ -477,9 +460,13 @@ public IImmutableList<string> GetTocFileKeySet(string key)
 
     public void RegisterToc(string tocFileKey, string fileKey)
     {
+#if NET7_0_OR_GREATER
+        ArgumentNullException.ThrowIfNullOrEmpty(fileKey);
```

Comment:
nit: it feels a bit over engineering of having to use a conditional compilation flag. Maybe we can consider adopting `ThrowIfNullOrEmpty` wait until .NET 6 official sunset in [November 12, 2024].